testing/sdk_login.py

```diff
@@ -0,0 +1,75 @@
+__CLI_LOGIN_EE_USERNAME = 'bootstrapuser'
+__CLI_LOGIN_EE_PASSWORD = 'deleteme'
+
+log = logging.getLogger(__name__)
+
+
+def login(dcosurl: str, username: str, password: str, is_enterprise: bool) -> str:
+    if is_enterprise:
+        log.info('logging into {} as {}'.format(dcosurl, username))
+        payload = {'uid': username, 'password': password}
+    else:
+        log.info('logging into {} with default open token'.format(dcosurl))
+        payload = {'token': __CLI_LOGIN_OPEN_TOKEN}
+
+    headers = {'Content-Type': 'application/json',}
+    login_endpoint = '{dcosurl}/acs/api/v1/auth/login'.format(dcosurl=dcosurl)
+    r = requests.post(login_endpoint, headers=headers, json=payload, verify=False)
+    assert r.status_code == 200, '{} failed {}: {}'.format(login_endpoint, r.status_code, r.text)
+
+    return r.json()['token']
+
+
+def configure_cli(dcosurl: str, token: str) -> None:
+    out, err, rc = shakedown.run_dcos_command('config set core.dcos_url {}'.format(dcosurl))
+    assert not rc, 'Failed to set core.dcos_url: {}'.format(err)
+    out, err, rc = shakedown.run_dcos_command('config set core.ssl_verify false')
+    assert not rc, 'Failed to set core.ssl_verify: {}'.format(err)
+    out, err, rc = shakedown.run_dcos_command('config set core.dcos_acs_token {}'.format(token))
+    assert not rc, 'Failed to set core.dcos_acs_token: {}'.format(err)
+
+
+def logout(dcosurl: str):
+    pass
+
+
+def login_session() -> None:
+    """Login to DC/OS.
+
+    Behavior is determined by the following environment variables:
+    CLUSTER_URL: full URL to the test cluster
+    DCOS_LOGIN_USERNAME: the EE user (defaults to bootstrapuser)
+    DCOS_LOGIN_PASSWORD: the EE password (defaults to deleteme)
+    DCOS_ENTERPRISE: determine how to authenticate (defaults to false)
+    DCOS_ACS_TOKEN: bypass auth and use the user supplied token
+
+    This should generally be used as a fixture in a framework's conftest.py:
+
+    @pytest.fixture(scope='session')
+    def configure_login():
+        yield from sdk_login.login_session()
+    """
+    try:
+        cluster_url = os.environ.get('CLUSTER_URL')
+        dcos_login_username = os.environ.get('DCOS_LOGIN_USERNAME', __CLI_LOGIN_EE_USERNAME)
+        dcos_login_password = os.environ.get('DCOS_LOGIN_PASSWORD', __CLI_LOGIN_EE_PASSWORD)
+        dcos_enterprise = os.environ.get('DCOS_ENTERPRISE', False) and True
```

Comment:
Yes, it will. Because it's a boolean, the idiom I was going for was "if the variable is defined," not the value it's set to. So things like case and "1" vs "True" vs "Yes" didn't have to be worried about. Either way needs to be made more clear.